src/commands/reset.js

```diff
@@ -8,37 +10,90 @@ class ResetCommand extends BaseCommand {
   /**
    * @param {Object} args
    * @param {Object} flags
-   * @param {DockerCompose} dockerCompose
+   * @param {resetSystemConfig} resetSystemConfig
+   * @param {checkSystemConfig} checkSystemConfig
    * @param {Config} config
+   * @param {ConfigCollection} configCollection
+   * @param {DockerCompose} dockerCompose
+   * @param {Docker} docker
+   * @param {tenderdashInitTask} tenderdashInitTask
    * @return {Promise<void>}
    */
   async runWithDependencies(
     args,
     {
       verbose: isVerbose,
+      hard: isHardReset,
+      'platform-only': isPlatformOnlyReset,
     },
-    dockerCompose,
+    resetSystemConfig,
+    checkSystemConfig,
     config,
+    configCollection,
+    dockerCompose,
+    docker,
+    tenderdashInitTask,
   ) {
     const tasks = new Listr([
       {
-        title: 'Reset node data',
+        title: 'Stop services',
+        task: async () => dockerCompose.stop(config.toEnvs()),
+      },
+      {
+        title: 'Reset platform data',
+        enabled: () => isPlatformOnlyReset,
         task: () => (
           new Listr([
             {
-              title: 'Remove Docker containers and associated data',
-              task: async () => dockerCompose.down(config.toEnvs()),
+              title: 'Remove platform services and associated data',
+              task: async () => {
+                const coreContainerNames = ['core', 'sentinel'];
+                const containerNames = await dockerCompose
+                  .getContainersList(config.toEnvs(), undefined, true);
+                const platformContainerNames = containerNames
+                  .filter((containerName) => !coreContainerNames.includes(containerName));
+
+                await dockerCompose.rm(config.toEnvs(), platformContainerNames);
+              },
+            },
+            {
+              title: 'Clean up platform volumes',
+              task: async () => docker.pruneVolumes(),
             },
           ])
         ),
       },
+      {
+        title: 'Remove all services and associated data',
+        enabled: () => !isPlatformOnlyReset,
+        task: async () => dockerCompose.down(config.toEnvs()),
+      },
+      {
+        title: `Reset config ${config.getName()}`,
+        enabled: () => isHardReset,
+        task: async (ctx, task) => {
+          if (checkSystemConfig(config.getName())) {
+            resetSystemConfig(configCollection, config.getName(), isPlatformOnlyReset);
+          } else {
+            // eslint-disable-next-line no-param-reassign
+            task.skip('config reset failed, only system configs can be reset');
```

Comment:
Hm.. It's not actually failed. If we want to fail it we need to call `task.error` or throw an exception. Maybe we just need to check is `--hard` present and config is not system then throw error even before running listr?

Reply:
I've changed the wording to "skipped" instead of "failed". But I guess the decision we need to make is if using `--hard` with a non-system config should just skip the config reset task (but still remove the containers and volumes), or if it should fail with an error and not do any tasks. My thinking is it is just part of the command that isn't possible, so skipping and emitting a message with `task.skip()` makes more sense than failing the whole command.

Reply:
Yeah, agree. But you consider it from the perspective of options validation. You logically can't do `--hard` with `--config=<not system>` so you preventing the whole thing because the input is not valid.

Reply:
Yeah, I changed my mind now. I think the behaviour you describe is more consistent with other commands, so I've changed the implementation now :+1: